packages/grid/_modules_/grid/components/columnHeaders/ColumnHeaderFilterIcon.tsx

```diff
@@ -0,0 +1,69 @@
+import { IconButton } from '@material-ui/core';
+import Badge from '@material-ui/core/Badge';
+import Tooltip from '@material-ui/core/Tooltip';
+import * as React from 'react';
+import { useGridSelector } from '../../hooks/features/core/useGridSelector';
+import { preferencePanelStateSelector } from '../../hooks/features/preferencesPanel/preferencePanelSelector';
+import { PreferencePanelsValue } from '../../hooks/features/preferencesPanel/preferencesPanelValue';
+import { useIcons } from '../../hooks/utils/useIcons';
+import { optionsSelector } from '../../hooks/utils/useOptionsProp';
+import { ApiContext } from '../api-context';
+
+export interface ColumnHeaderFilterIconProps {
+  counter?: number;
+}
+
+export const ColumnHeaderFilterIcon: React.FC<ColumnHeaderFilterIconProps> = (props) => {
+  const { counter } = props;
+  const apiRef = React.useContext(ApiContext);
+  const options = useGridSelector(apiRef, optionsSelector);
+  const preferencePanel = useGridSelector(apiRef, preferencePanelStateSelector);
+
+  const tooltipContentNode = React.useMemo(() => {
+    return <div>{counter} active filter(s)</div>;
+  }, [counter]);
+
+  const icons = useIcons();
+  const filteredColumnIconElement = React.createElement(icons.ColumnFiltered!, {
+    fontSize: 'small',
+  });
+  const toggleFilter = React.useCallback(
+    (event: React.MouseEvent<HTMLButtonElement>) => {
+      event.preventDefault();
+      event.stopPropagation();
+
+      const { open, openedPanelValue } = preferencePanel;
+      if (open && openedPanelValue === PreferencePanelsValue.filters) {
+        apiRef!.current.hideFilterPanel();
+      } else {
+        apiRef!.current.showFilterPanel();
+      }
+    },
+    [apiRef, preferencePanel],
+  );
+
+  if (!counter || options.disableColumnFilter || !options.hideToolbar) {
+    return null;
+  }
+
+  const iconButton = (
+    <IconButton onClick={toggleFilter} color="default" aria-label="Show Filters" size="small">
+      {filteredColumnIconElement}
+    </IconButton>
+  );
+
+  return (
+    <Tooltip title={tooltipContentNode} enterDelay={1000}>
+      <div className="MuiDataGrid-sortIconContainer">
+        <div>
```

Comment:
A specific reason for using two divs?

Reply:
Yes it keeps the alignment with the icon and the badge on numeric columns